backend/main.py

```
@@ -0,0 +1,191 @@
+                status=schemas.SpotStatus(crowd_level=schemas.CrowdLevel.HIGH, rating=3),
+                author=schemas.AuthorInfo(id=UUID("00000000-0000-0000-0000-000000000099"), username="admin"),
+                skin=schemas.SkinInfo(id=UUID("00000000-0000-0000-0000-000000000001"), name="Default Pin", image_url="https://via.placeholder.com/50")
+            )
+
+        # from memory DB
+        found_spot = next((s for s in fake_spots_db if s.id == spot_id), None)
+        if not found_spot:
+            raise HTTPException(status_code=404, detail="Spot not found")
+        
+        return found_spot
+    return found_spot
```

Ruff (0.14.7)

Undefined name `found_spot`

(F821)